transports/ftp/src/main/java/org/mule/transport/ftp/FtpConnector.java

```diff
@@ -543,29 +548,50 @@ public void setupFileType(FTPClient client, ImmutableEndpoint endpoint) throws E
      *         does not support streaming
      */
     @Override
-    public OutputStream getOutputStream(OutboundEndpoint endpoint, MuleEvent event) throws MuleException
+    public OutputStream getOutputStream(final OutboundEndpoint endpoint, MuleEvent event) throws MuleException
     {
         try
         {
             final EndpointURI uri = endpoint.getEndpointURI();
             String filename = getFilename(endpoint, event.getMessage());
 
-            final FTPClient client;
+            final FTPClient[] client = {null};
+            RetryCallback callbackReconnection = new RetryCallback()
+            {
+
+                @Override
+                public void doWork(RetryContext context) throws Exception
+                {
+                    client[0] = createFtpClient(endpoint);
+                }
+
+                @Override
+                public String getWorkDescription()
+                {
+                    return getConnectionDescription();
+                }
+
+                @Override
+                public Connector getWorkOwner()
+                {
+                    return FtpConnector.this;
+                }
+            };
             try
             {
-                client = this.createFtpClient(endpoint);
+                getRetryPolicyTemplate().execute(callbackReconnection, muleContext.getWorkManager());
             }
-            catch (Exception e)
+            catch (RetryPolicyExhaustedException retryPolicyExhaustedException)
             {
-                throw new ConnectException(e, this);
+                throw retryPolicyExhaustedException;
             }
 
             try
             {
-                OutputStream out = client.storeFileStream(filename);
+                OutputStream out = client[0].storeFileStream(filename);
```

Comment:
Is ugly to have this code having to access the array. Looks like this try block can be extracted into an utility method receiving the client as a parameter